### ISSUE TITLE
NWB files without cached namespaces do not validate

### ISSUE THREAD

**Issue by rly** [NeurodataWithoutBorders/pynwb#1157]
Example: The Allen Institute Visual Coding NWB files do not have cached specs.
```
python -m pynwb.validate probe_760640083_lfp.nwb
```
Output:
```
c:\users\ryan\documents\nwb\hdmf\src\hdmf\backends\hdf5\h5tools.py:100: UserWarning: No cached namespaces found in C:\Users\Ryan\Documents\NWB\allen-visual-coding-nwb\visual_coding\example_ecephys_project_cache\session_756029989\probe_760640083_lfp.nwb
  warnings.warn(msg)
The file C:\Users\Ryan\Documents\NWB\allen-visual-coding-nwb\visual_coding\example_ecephys_project_cache\session_756029989\probe_760640083_lfp.nwb has no cached namespace information. Falling back to pynwb namespace information.
Validating C:\Users\Ryan\Documents\NWB\allen-visual-coding-nwb\visual_coding\example_ecephys_project_cache\session_756029989\probe_760640083_lfp.nwb against pynwb namespace information using namespace hdmf-common.
Traceback (most recent call last):
  File "C:\Users\Ryan\Miniconda3\envs\pynwb-dev\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Ryan\Miniconda3\envs\pynwb-dev\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\users\ryan\documents\nwb\pynwb\src\pynwb\validate.py", line 115, in <module>
    main()
  File "c:\users\ryan\documents\nwb\pynwb\src\pynwb\validate.py", line 109, in main
    ret = ret or _validate_helper(io=io, namespace=ns)
  File "c:\users\ryan\documents\nwb\pynwb\src\pynwb\validate.py", line 24, in _validate_helper
    errors = validate(**kwargs)
  File "c:\users\ryan\documents\nwb\hdmf\src\hdmf\utils.py", line 455, in func_call
    return func(**parsed['args'])
  File "c:\users\ryan\documents\nwb\pynwb\src\pynwb\__init__.py", line 199, in validate
    return validator.validate(builder)
  File "c:\users\ryan\documents\nwb\hdmf\src\hdmf\utils.py", line 453, in func_call
    return func(args[0], **parsed['args'])
  File "c:\users\ryan\documents\nwb\hdmf\src\hdmf\validate\validator.py", line 233, in validate
    raise ValueError(msg)
ValueError: builder for container 'root' must have data type defined with attribute 'data_type'
```

It looks like when there are no cached specs, the validator tries to validate the file against the hdmf-common-schema, but the file uses the data type key `'neurodata_type'` instead of `'data_type'`.

**Comment by rly** [comment]
Using the default validate function, however, seems to work:
```python
from pynwb import validate, NWBHDF5IO
with NWBHDF5IO('probe_760640083_lfp.nwb') as io:
    io.validate()
```
output:
```
[TimeSeries/control (processing/current_source_density/current_source_density/control): incorrect shape - expected '[None]', got '(384, 2)', 
DynamicTable/colnames (general/extracellular_ephys/electrodes.colnames): incorrect type - expected 'text', got 'ascii']
```

**Comment by t-b** [comment]
@rly Does it work with https://github.com/NeurodataWithoutBorders/pynwb/pull/1094?

We should add some tests using the CLI validation tool as well into `test.py`. This seems to work too differently.

**Comment by rly** [comment]
@t-b No, it does not work with #1094. Testing #1094 led me here.

**Comment by t-b** [comment]
This should be fixed with #1164.
